### ISSUE TITLE
EncodedEventId::operator()() is NOT const

### ISSUE THREAD

**Issue by VinInn** [cms-sw/cmssw#33460]
```
/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_0_X_2021-04-18-0000/src/SimDataFormats/EncodedEventId/interface/EncodedEventId.h:31:12: note: candidate: 'uint32_t EncodedEventId::operator()()' <near match>
   31 |   uint32_t operator()() { return id_; }
      |            ^~~~~~~~
/cvmfs/cms-ib.cern.ch/week1/slc7_amd64_gcc900/cms/cmssw/CMSSW_12_0_X_2021-04-18-0000/src/SimDataFormats/EncodedEventId/interface/EncodedEventId.h:31:12: note:   passing 'const EncodedEventId*' as 'this' argument discards qualifiers
```

one is forced to use ```rawId```.

**Comment by makortel** [comment]
assign simulation

**Comment by civanch** [comment]
+1

fixed in #33506
